thunder/executors/torch_autograd.py

```diff
@@ -259,21 +314,36 @@ def split_forward_backward(computation_trc: TraceCtx, compile_data, compile_stat
         # NOTE: `_transformer_engine_bwd_fp8_meta_sync` may mutate `fw_extrace` or `bw_extrace`.
         _transformer_engine_bwd_fp8_meta_sync(fw_extrace, bw_extrace)
 
-    fw_extrace = del_last_used(fw_extrace)
-    fw_traces.append(fw_extrace)
-
     bw_extrace = del_last_used(bw_extrace, clear_mutable_collections=True)
+    bw_extrace = rename_bwd_trace_outputs(bw_extrace, fw_extrace)
+    # We only want the forward function to be called with `te.fp8_autocast` manager.
+    bw_extrace._include_te_fp8_autocast = False
     bw_traces.append(bw_extrace)
 
-    bw_trace = rename_bwd_trace_outputs(bw_extrace, fw_extrace)
+    for transform in compile_data.post_optimization_transforms:
+        utils.check_type(transform, PostOptimizationTransform)
+        bw_extrace = transform.transform_trace(bw_extrace, executors_list=compile_data.executors_list)
+        bw_traces.append(bw_extrace)
+
+    # Update the forward trace with the correct backward function
+    if any(bsym.sym.name == "connect_to_autograd_impl" for bsym in reversed(fw_extrace.bound_symbols)):
```

Comment:
Why could this not be `bsym.sym == connect_to_autograd`?
(I guess it's two questions, one why it is impl and why we compare the name rather than the sym.)

Reply:
> Why could this not be bsym.sym == connect_to_autograd?

At this stage "transform for execution" phase has replaced Thunder's general symbol `connect_to_autograd` which works on TensorProxy with PyTorch executor-specific symbol `connect_to_autograd_impl` that works with torch.Tensors.

As for string comparison, I probably didn't want to import anything from `torch.executors.torchex` due to possible cyclic dependencies. I'll revisit this.

Reply:
If needed, importing in the function should work and be preferable to string comparison.